DYN-9195: Net 10.0 update

dnenov opened the pull request:
### Purpose

This PR is part of the `dotnet 10.0` upgrade initiative for Dynamo and all supporting components. 

The primary change in this transition is the removal of BinaryFormatter, which was previously used to deserialize embedded resources. Because the sample packages did not include any resources, it was difficult to evaluate the impact of this change.

To address this, I added a custom node icon to the original `.NET 8.0` version and then implemented the proper handling for it in `.NET 10.0`. This ensures that the new approach to resource management is correctly demonstrated and validated.

Addresses https://jira.autodesk.com/browse/DYN-9195 part of https://jira.autodesk.com/browse/DYN-9194 epic.

### Declarations

Check these if you believe they are true

- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB
- [ ] This PR introduces new feature code involve network connecting and is tested with no-network mode.

### Release Notes

- `net10.0` upgrade changes 
- included an example of a custom icon to a Dynamo packge node

### Reviewers

@saintentropy 
@zeusongit 
@QilongTang 
@jasonstratton 

### FYIs

@achintyabhat

dnenov committed:
add node icon example

- adding a simple icon to a node to demonstrate the 8.0 workflow to contrast and show a fix when introducing 10.0

dnenov committed:
image to png update

- now the image is added as a File and in .png format
- this alone works in order to correctly provide the icon to Dynamo

dnenov committed:
.net 10.0 upgrade

- updated all projects to target .net 10.0
- updated workflow yaml
- updated readme reference to dotnet

dnenov committed:
add icon resources to build directory

- add customisation.dll file to the build output directory

dnenov committed:
revert formatting changes

QilongTang commented:
@dnenov Check out the latest Nuget after your PR merge, here: https://www.nuget.org/packages/DynamoVisualProgramming.Core/4.0.0-beta2526. Curious if this can be applied to any of the dependency in this repo

dnenov committed:
nuget update

- updated nuget package references

dnenov commented:
> @dnenov Check out the latest Nuget after your PR merge, here: https://www.nuget.org/packages/DynamoVisualProgramming.Core/4.0.0-beta2526. Curious if this can be applied to any of the dependency in this repo

All nugets updated now @QilongTang! It is building fine and loading as before.

QilongTang commented:
We will wait for newer Nugets with actual .NET10 built binaries before merging

zeusongit commented:
> We will wait for newer Nugets with actual .NET10 built binaries before merging

We should update this now, as we have .Net10 nugets available?

dnenov committed:
nugets bump to 2685

dnenov committed:
remove redundant file

dnenov committed:
removed resource custom tool

dnenov commented:
> > We will wait for newer Nugets with actual .NET10 built binaries before merging
> 
> We should update this now, as we have .Net10 nugets available?

Updated all nugets to `4.0.0-beta2685`

zeusongit committed:
Update build.yml

QilongTang commented:
@dnenov Please also touch https://github.com/DynamoDS/DynamoSamples/blob/master/.github/workflows/build.yml to make the build check pass

dnenov committed:
update to nugs, yml

- update the dotnet-version: "10.0.100-preview.5.25277.114"
- update to latest nugets

dnenov commented:
> @dnenov Please also touch https://github.com/DynamoDS/DynamoSamples/blob/master/.github/workflows/build.yml to make the build check pass

I updated the `dotnet version` now, and also update to the latest `nuget` packages!
